Mathlib/Data/Set/Subset.lean

```diff
@@ -0,0 +1,260 @@
+/-
+Copyright (c) 2024 Miguel Marco. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Miguel Marco
+-/
+import Mathlib.Data.Set.Function
+import Mathlib.Data.Set.Functor
+import Mathlib.Lean.Expr.ExtraRecognizers
+
+/-!
+# Sets in subtypes
+
+This file defines notation for sets in a type pulled down to sets in a subtype, and sets in
+a subtype, lifted up to sets in the ambient type.
+
+It also provides some related lemmas for convenience.
+
+## Notation
+
+Let `α` be a `Type`, `A B : Set α` two sets in `α`, and `C : Set ↑A` a set in the subtype `↑A`.
+
+- `A ↓∩ B` denotes `(Subtype.val ⁻¹' B : Set A)` (that is, `{x : ↑A | ↑x ∈ B}`).
+- `↑C` denotes `Subtype.val '' C` (that is, `{x : α | ∃ y ∈ C, ↑y = x}`).
+
+To access this notation, type `open Subset`.
+
+## Tags
+
+subsets
+-/
+
+open Set
+
+universe u
+
+variable {α β : Type u} {A B C: Set α} {D E : Set ↑A}
```

Comment:
```suggestion
variable {α β : Type u} {A B C : Set α} {D E : Set ↑A}
```
Note: you can write `Set A` instead of `Set ↑A`. It's more common to rely on the auto-inserted coercion. No need to change it though, just mentioning it.